ml_garden/core/steps/calculate_metrics.py

```diff
@@ -50,6 +60,56 @@ def _calculate_metrics(self, true_values: pd.Series, predictions: pd.Series) ->
             "Median Absolute Error": str(median_absolute_error),
         }
 
+    def _calculate_classification_metrics(
+        self, true_values: pd.Series, predictions: pd.Series
+    ) -> dict:
+        """Calculate classification metrics.
+
+        Parameters
+        ----------
+        true_values : pd.Series
+            True values
+        predictions : pd.Series
+            Predicted class labels
+
+        Returns
+        -------
+        dict
+            Classification metrics
+        """
+        accuracy = accuracy_score(true_values, predictions)
+
+        # Calculate per-class precision, recall, and f1-score
+        precision, recall, f1, support = precision_recall_fscore_support(true_values, predictions)
+
+        # Overall weighted averages
+        weighted_precision = precision_score(true_values, predictions, average="weighted")
+        weighted_recall = recall_score(true_values, predictions, average="weighted")
+        weighted_f1 = f1_score(true_values, predictions, average="weighted")
+
+        cm = confusion_matrix(true_values, predictions)
+
+        # Create a dictionary for per-class metrics
+        class_metrics = {}
+        for i, class_label in enumerate(np.unique(true_values)):
+            class_metrics[f"Class_{class_label}"] = {
+                "Precision": str(precision[i]),
+                "Recall": str(recall[i]),
+                "F1 Score": str(f1[i]),
+                "Support": str(support[i]),
+            }
+
+        return {
+            "Overall": {
+                "Accuracy": str(accuracy),
+                "Weighted Precision": str(weighted_precision),
+                "Weighted Recall": str(weighted_recall),
+                "Weighted F1 Score": str(weighted_f1),
+            },
+            "Per_Class": class_metrics,
+            "Confusion Matrix": str(cm.tolist()),
+        }
```

Comment:
We _could_ use `TypedDict`s or `dataclasses` here so that the structure of the metrics is self-documenting.